src/txn/txn_rollback_to_stable.c

```diff
@@ -809,6 +814,63 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __rollback_to_stable_btree_hs_cleanup --
+ *     Wipe all history store updates for the selected btrees (non-timestamped tables)
+ */
+static int
+__rollback_to_stable_btree_hs_cleanup(WT_SESSION_IMPL *session, uint8_t *rollback_bitstring)
+{
+    WT_CURSOR *hs_cursor;
+    WT_CURSOR_BTREE *cbt;
+    WT_DECL_ITEM(hs_key);
+    WT_DECL_RET;
+    WT_TIME_PAIR hs_start, hs_stop;
+    WT_UPDATE *hs_upd;
+    uint32_t hs_btree_id, session_flags;
+
+    hs_cursor = NULL;
+    hs_upd = NULL;
+    session_flags = 0;
+
+    /* Allocate buffers for the history store key. */
+    WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
+
+    /* Open a history store table cursor. */
+    WT_ERR(__wt_hs_cursor(session, &session_flags));
+    hs_cursor = session->hs_cursor;
+    cbt = (WT_CURSOR_BTREE *)hs_cursor;
+
+    /* Walk the file. */
+    while ((ret = hs_cursor->next(hs_cursor)) == 0) {
```

Comment:
Does this do a sequential walk through the entire history store table?
Could we instead search specifically for the btrees we need by ID, and just traverse those records? It makes this code more complex, but means we don't need to walk entries with timestamps to find those we are interested in here.

Reply:
I fixed it using the specific btree instead of scanning full history store. Also changed to do the cleanup whenever a non-timestamp table is found in the metadata scan.